### PR TITLE
rename query fix

### PR DESCRIPTION
**Pull request by xannem**
## Summary

- [x] Removing `rename query` from additional options and using behavior of double-clicking title instead
- [x] Fixing regression bug of not having query title displayed 
- [x] Changing searching validation for duplicate query name to only include queries that are created by user 

Kapture 2023-05-04 at 15 43 23

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query-builder/style/shared/_query-loader.scss [hunk 1]
```diff
@@ -243,6 +244,7 @@
 
     &__author__icon {
       svg {
+        display: flex;
```
Comment (akphi) [inline packages/legend-query-builder/style/shared/_query-loader.scss:247]: this looks weird, don't think we ever need to set flex to the `svg`

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -540,15 +541,6 @@ const QueryEditorHeaderContent = observer(
             disabled={editorStore.isViewProjectActionDisabled}
             content={
               <MenuContent>
-                {/* TODO: remove this and move it to inline editing directly on the title */}
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:543]: It's okay to leave this here I think, let's remove the comment

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 3]
```diff
@@ -416,6 +416,7 @@ const QueryEditorHeaderContent = observer(
           </div>
         ) : (
           <div
+            onDoubleClick={updateQuery}
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:419]: nice! But in terms of UX, we need visual hints that this is possible, using tooltip or some CSS border, or altering the `cursor: pointer`

**Review by gayathrir11 — APPROVED** [review]

File: packages/legend-application-query/style/_query-editor.scss [hunk 4]
```diff
@@ -61,6 +61,10 @@
         padding: 0.5rem;
         margin-right: 10%;
       }
+
+      &__title:hover {
```
Comment (akphi) [inline packages/legend-application-query/style/_query-editor.scss:65]: usually, when one sees something like this, it means you can set `cursor: pointer` on the base CSS class

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 5]
```diff
@@ -384,15 +391,24 @@ const QueryEditorHeaderContent = observer(
...
-                  onKeyDown={onEnter}
+                  onKeyDown={(event) => {
+                    if (event.code === 'Enter') {
+                      event.stopPropagation();
+                      setShowQueryNameEditInput(false);
+                      onEnter(event);
+                    } else if (event.code === 'Escape') {
+                      setShowQueryNameEditInput(false);
+                      event.stopPropagation();
+                      renameState.renameQueryState.reset();
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:407]: Hitting `Esc` should do select all, isntead of erasing everything - `renameRef.current?.select();`

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 6]
```diff
@@ -408,16 +424,17 @@ const QueryEditorHeaderContent = observer(
...
                 <CheckIcon />
               </button>
             </PanelListItem>
           </div>
         ) : (
           <div
+            onDoubleClick={updateQuery}
             className="query-editor__header__content__main query-editor__header__content__title"
-            title="Query title"
+            title="Double-click to rename Query"
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:437]:
```suggestion
            title="Double-click to rename query"
```

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 7]
```diff
@@ -408,16 +424,17 @@ const QueryEditorHeaderContent = observer(
                   'btn--icon__caution': isExistingQueryName,
                 })}
                 onClick={renameQuery}
-                title="Rename Query"
+                title="Rename query"
               >
                 <CheckIcon />
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:429]: You know, I think we can do a little better, we don't need the check button, on `Enter`, we will save, I think this is a fairly clear interaction

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 8]
```diff
@@ -384,15 +391,24 @@ const QueryEditorHeaderContent = observer(
...
+                      setShowQueryNameEditInput(false);
+                      event.stopPropagation();
+                      renameState.renameQueryState.reset();
+                      renameState.editorStore.setRenameState(undefined);
+                    }
+                  }}
                   value={renameState.queryName}
-                  placeholder="Search"
                 />
                 {isExistingQueryName && (
```
Comment (akphi) [inline packages/legend-application-query/src/components/QueryEditor.tsx:413]: Did we leverage the API you added in https://github.com/finos/legend-engine/pull/1669 ?

File: packages/legend-query-builder/src/stores/QueryLoaderState.ts [hunk 9]
```diff
@@ -246,7 +246,7 @@ export class QueryLoaderState {
         name,
       )) as Query;
       this.onQueryRenamed?.(query);
-      this.applicationStore.notificationService.notify(
+      this.applicationStore.notificationService.notifySuccess(
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryLoaderState.ts:249]: TODO: just making a comment here, probably would be great if we can bring the same duplicated name warning here

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-query-builder/src/stores/QueryLoaderState.ts (hunk 9 above)
Comment (xannem) [reply packages/legend-query-builder/src/stores/QueryLoaderState.ts:249]: Makes sense, added 👍

**Review by akphi — APPROVED** [review]

**Pull request merged**